### PR TITLE
Let non-admin users remove themselves from orgs and projects

### PR DESCRIPTION
**Pull request by begelundmuller**
Closes issue reported in [Slack](https://rilldata.slack.com/archives/CTZ8XBQ85/p1713218977194649).

### CODE REVIEW

File: admin/server/organizations.go [hunk 1]
```diff
@@ -415,52 +415,64 @@ func (s *Server) RemoveOrganizationMember(ctx context.Context, req *adminv1.Remo
...
 		}
+
 		return &adminv1.RemoveOrganizationMemberResponse{}, nil
 	}
 
+	// The caller must either have ManageOrgMembers permission or be the user being removed.
+	claims := auth.GetClaims(ctx)
+	allow := false
+	allow = allow || claims.OrganizationPermissions(ctx, org.ID).ManageOrgMembers
+	allow = allow || (claims.OwnerType() == auth.OwnerTypeUser && claims.OwnerID() == user.ID)
```
Comment (pjain1) [inline admin/server/organizations.go:452]:
```suggestion
	allow := claims.ProjectPermissions(ctx, proj.OrganizationID, proj.ID).ManageProjectMembers || (claims.OwnerType() == auth.OwnerTypeUser && claims.OwnerID() == user.ID)
```

File: admin/server/projects.go [hunk 2]
```diff
@@ -711,31 +711,44 @@ func (s *Server) RemoveProjectMember(ctx context.Context, req *adminv1.RemovePro
...
 			return nil, status.Error(codes.Internal, err.Error())
 		}
 		return &adminv1.RemoveProjectMemberResponse{}, nil
 	}
 
+	// The caller must either have ManageProjectMembers permission or be the user being removed.
+	claims := auth.GetClaims(ctx)
+	allow := false
+	allow = allow || claims.ProjectPermissions(ctx, proj.OrganizationID, proj.ID).ManageProjectMembers
+	allow = allow || (claims.OwnerType() == auth.OwnerTypeUser && claims.OwnerID() == user.ID)
```
Comment (pjain1) [inline admin/server/projects.go:747]:
```suggestion
	allow := claims.ProjectPermissions(ctx, proj.OrganizationID, proj.ID).ManageProjectMembers || (claims.OwnerType() == auth.OwnerTypeUser && claims.OwnerID() == user.ID)
```

File: admin/server/organizations.go (hunk 1 above)
Comment (pjain1) [inline admin/server/organizations.go:452]:
```suggestion
	allow := claims.OrganizationPermissions(ctx, org.ID).ManageOrgMembers || (claims.OwnerType() == auth.OwnerTypeUser && claims.OwnerID() == user.ID)
```

**Review by pjain1 — APPROVED** [review]

File: admin/server/projects.go (hunk 2 above)
Comment (begelundmuller) [reply admin/server/projects.go]: Google's Go style guide recommends breaking complex conditionals into multiple lines: https://google.github.io/styleguide/go/guide#maintainability.

I refactored it to use separate vars instead:
```go
	isManager := claims.OrganizationPermissions(ctx, org.ID).ManageOrgMembers
	isSelf := claims.OwnerType() == auth.OwnerTypeUser && claims.OwnerID() == user.ID
	if !isManager && !isSelf {
```

File: admin/server/organizations.go (hunk 1 above)
Comment (begelundmuller) [reply admin/server/organizations.go]: See my response on the other review comment

File: admin/server/projects.go (hunk 2 above)
Comment (pjain1) [reply admin/server/projects.go]: ok

**Pull request merged**